command/server/config.go

```diff
@@ -997,6 +1015,106 @@ func ParseStorage(result *Config, list *ast.ObjectList, name string) error {
 	return nil
 }
 
+// storageAddressKeys is a maps a storage backend type to its associated
+// that may configuration whose values are URLs, IP addresses, or host:port
+// style addresses. All physical storage types must have an entry in this map,
+// otherwise our normalization check will fail when parsing the storage entry
+// config. Physical storage types which don't contain such keys should include
+// an empty array.
+var storageAddressKeys = map[string][]string{
```

Comment:
I'm guessing the risk of drift with this config map is small, since we're not in the habit of adding storage backends regularly or changing their options. But if we do add more in the future, will we need to remember to update this or will there be some kind of failure that points to this needing to be updated?

Reply:
I was concerned about the same thing so I ultimately decided [to raise an error if an entry wasn't present in the map](https://github.com/hashicorp/vault/pull/29228/files#diff-0b1642f753dfe53bd60b0d3f731e4e7728f66e550b8820013a46e48db9c317b5R1060) which ultimately resulted in having to [change a few existing tests](https://github.com/hashicorp/vault/pull/29228/files#diff-cb056a97479a85b978d0244281f99f3a9c948aaaaeb1a91846382c64ceeac43eR194) to use known keys.

Reply:
Excellent! That's exactly what I was hoping for. Thank you!